Sources/SwiftSyntax/Documentation.docc/Macro Versioning.md

```diff
@@ -0,0 +1,29 @@
+# Updating a Macro to a New Major swift-syntax Version
+
+Guidance of how to version macros when a new major swift-syntax version is released.
+
+## Summary
+
+Update the minor version of a macro when updating its swift-syntax version dependency.
+
+## Detailed Explanation
+
+All the ideas described in the following apply to all packages that depend on swift-syntax, not only macros.
+
+For simplicity, this article assumes that `509` is the current swift-syntax version and `510` the next, but everything applies to any other major swift-syntax version update, including version jumps to `600`. 
+
+When staring to write a macro, no special considerations need to be made with regard in terms of versioning. Depending on swift-syntax as with `from: "509.0.0"` will make sure that the macro receives any bug fix updates to swift-syntax 509. For example, a macro depending on swift-syntax 509 might be released as version 1.0, 1.2, 2.0, … 
```

Comment:
```suggestion
When starting to write a macro, no special considerations need to be made with regard in terms of versioning. Depending on swift-syntax as with `from: "509.0.0"` will make sure that the macro receives any bug fix updates to swift-syntax 509. For example, a macro depending on swift-syntax 509 might be released as version 1.0, 1.2, 2.0, … 
```

Reply:
Thanks for catching these typos @tevelee